Using multiple `FILLER`  causes OutOfMemoryError

Using `FILLER` more than once in the same module (scope?) causes an infinite loop when reporting lints, making the LSP or the linter consume memory until an OOM occurs.

Program that prompts this below. It's not as simple as just redefining something with FILLER, something else has to be true, not sure what yet.

If you replace one or both of the `FILLER` with the equivalent length `A` field it all works fine (after kill -9 on the LSP and a restart)

```natural
* >Natural Source Header 000000
* :Mode S
* :CP
* <Natural Source Header
****************************************************************
DEFINE DATA LOCAL
01 #COMMAND (A250)
01 #ERROR (N4)
01 #MESSAGE (A72)
01 #LIBRARY (A8) INIT <'SYSMAIN'>
01 #INDEX (I2)
01 #INDEX-MOD (I2)
01 #SOURCE (A8)
01 #TARGET (A8)
01 #FNC (A1)
01 #GOOD-COUNT (I2) INIT <0>
01 #MISSING-COUNT (I2) INIT <0>
01 #ERR-COUNT (I2) INIT <0>
01 #CANDIDATES (A20/1:1) INIT <
'RUBBISH'
>
01 REDEFINE #CANDIDATES
  02 #CANDIDATE-DETAIL (1:41)
    03 #OBJECT (A8)
    03 FILLER   2X
    03 #STATUS (A6)
    03 FILLER   4X
END-DEFINE
*
INPUT 10X 'QUARANTINE PROGRAM' /
      10X '------------------' //
      10X 'PRINT LIST     (P)' /
      10X 'QUARANTINE     (Q)' /
      10X 'ROLLBACK       (R)' /
      10X 'EXIT           (.)' //
      10X 'CHOOSE: ' #FNC
*
DECIDE ON EVERY VALUE OF #FNC
  VALUE 'P'
    PERFORM PRINT-MODULES
  VALUE 'Q'
    #SOURCE := 'MAINLIBR'
    #TARGET := 'QUARHOLE'
    PERFORM MOVE-MODULES
  VALUE 'R'
    #SOURCE := 'QUARHOLE'
    #TARGET := 'MAINLIBR'
    PERFORM MOVE-MODULES
  VALUE '.'
    STOP
  NONE
    REINPUT 'Try again.' MARK *#FNC
END-DECIDE
*
DEFINE SUBROUTINE PRINT-MODULES
  DISPLAY #CANDIDATES(*)
END-SUBROUTINE
*
DEFINE SUBROUTINE MOVE-MODULES
  FOR #INDEX 1 TO *UBOUND(#CANDIDATES)
    COMPRESS 'MOVE ALL' #OBJECT(#INDEX) 'FM' #SOURCE 'TO' #TARGET
             'WITH REPLACE NOPROMPT' INTO #COMMAND
    CALLNAT 'MAINUSER' #COMMAND #ERROR #MESSAGE #LIBRARY
    DECIDE ON FIRST VALUE OF #ERROR
      VALUE 4810
        #ERR-COUNT := #ERR-COUNT + 1
        #STATUS(#INDEX) := 'FAILED'
      VALUE 4868
        #GOOD-COUNT := #GOOD-COUNT + 1
        #STATUS(#INDEX) := 'MOVED '
      VALUE 4867
        #MISSING-COUNT := #MISSING-COUNT + 1
        #STATUS(#INDEX) := 'ABSENT'
      NONE VALUE
        WRITE #ERROR #MESSAGE
    END-DECIDE
    /* Need to pause and display something every 8 loops or you get err 1009
    /* from MAINUSER
    #INDEX-MOD := #INDEX
    DIVIDE 8 INTO #INDEX-MOD REMAINDER #INDEX-MOD
    IF #INDEX-MOD = 0 THEN
      WRITE "PROCESSED" #INDEX "HIT ENTER"
    END-IF
  END-FOR
  DISPLAY #CANDIDATE-DETAIL(*)
  WRITE   10X 'MOVED    :' #GOOD-COUNT /
          10X 'FAILED   :' #ERR-COUNT /
          10X 'MISSING  :' #MISSING-COUNT /
          10X '----------' /
          10X 'OF TOTAL :' *UBOUND(#CANDIDATES)
END-SUBROUTINE
END

```


---

~Not sure what's going on here, it worked before (but we have aggregated a bunch of source into the same lib). This is for about 5,500 objects.~

```
Exception in thread "main" java.lang.OutOfMemoryError
        at java.base/jdk.internal.reflect.DirectConstructorHandleAccessor.newInstance(DirectConstructorHandleAccessor.java:62)
        at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:502)
        at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:486)
        at java.base/java.util.concurrent.ForkJoinTask.getThrowableException(ForkJoinTask.java:542)
        at java.base/java.util.concurrent.ForkJoinTask.reportException(ForkJoinTask.java:567)
        at java.base/java.util.concurrent.ForkJoinTask.invoke(ForkJoinTask.java:670)
        at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateParallel(ForEachOps.java:160)
        at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateParallel(ForEachOps.java:174)
        at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:233)
        at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:596)
        at java.base/java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:765)
        at org.amshove.natlint.cli.CliAnalyzer.analyze(CliAnalyzer.java:112)
        at org.amshove.natlint.cli.CliAnalyzer.run(CliAnalyzer.java:90)
        at org.amshove.natlint.cli.AnalyzeCommand.call(AnalyzeCommand.java:99)
        at org.amshove.natlint.cli.AnalyzeCommand.call(AnalyzeCommand.java:19)
        at picocli.CommandLine.executeUserObject(CommandLine.java:1953)
        at picocli.CommandLine.access$1300(CommandLine.java:145)
        at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2358)
        at picocli.CommandLine$RunLast.handle(CommandLine.java:2352)
        at picocli.CommandLine$RunLast.handle(CommandLine.java:2314)
        at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2179)
        at picocli.CommandLine$RunLast.execute(CommandLine.java:2316)
        at picocli.CommandLine.execute(CommandLine.java:2078)
        at org.amshove.natlint.App.main(App.java:10)
Caused by: java.lang.OutOfMemoryError: Java heap space: failed reallocation of scalar replaced objects
        at java.base/java.util.Arrays.copyOf(Arrays.java:3541)
        at java.base/java.lang.AbstractStringBuilder.ensureCapacityInternal(AbstractStringBuilder.java:242)
        at java.base/java.lang.AbstractStringBuilder.append(AbstractStringBuilder.java:686)
        at java.base/java.lang.StringBuilder.append(StringBuilder.java:218)
        at java.base/java.lang.StringBuilder.append(StringBuilder.java:91)
        at java.base/java.util.Formatter$FixedString.print(Formatter.java:2876)
        at java.base/java.util.Formatter.format(Formatter.java:2780)
        at java.base/java.util.Formatter.format(Formatter.java:2728)
        at java.base/java.lang.String.formatted(String.java:4452)
        at org.amshove.natparse.parsing.ParserErrors.unexpectedToken(ParserErrors.java:74)
        at org.amshove.natparse.parsing.AbstractParser.consume(AbstractParser.java:228)
        at org.amshove.natparse.parsing.DefineDataParser.parseRedefineFiller(DefineDataParser.java:501)
        at org.amshove.natparse.parsing.DefineDataParser.scope(DefineDataParser.java:178)
        at org.amshove.natparse.parsing.DefineDataParser.dataDefinition(DefineDataParser.java:133)
        at org.amshove.natparse.parsing.DefineDataParser.parseInternal(DefineDataParser.java:60)
        at org.amshove.natparse.parsing.DefineDataParser.parseInternal(DefineDataParser.java:12)
        at org.amshove.natparse.parsing.AbstractParser.parse(AbstractParser.java:43)
        at org.amshove.natparse.parsing.DefineDataParser.parse(DefineDataParser.java:12)
        at org.amshove.natparse.parsing.NaturalParser.parseDefineData(NaturalParser.java:191)
        at org.amshove.natparse.parsing.NaturalParser.parseModule(NaturalParser.java:69)
        at org.amshove.natparse.parsing.NaturalParser.parse(NaturalParser.java:36)
        at org.amshove.natlint.cli.CliAnalyzer.parse(CliAnalyzer.java:280)
        at org.amshove.natlint.cli.CliAnalyzer.lambda$analyze$0(CliAnalyzer.java:141)
        at org.amshove.natlint.cli.CliAnalyzer$$Lambda/0x00007fcfc80b1748.accept(Unknown Source)
        at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:184)
        at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1708)
        at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
        at java.base/java.util.stream.ForEachOps$ForEachTask.compute(ForEachOps.java:291)
        at java.base/java.util.concurrent.CountedCompleter.exec(CountedCompleter.java:754)
        at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:387)
        at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1312)
        at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1843)
```

--- a/libs/natparse/src/main/java/org/amshove/natparse/parsing/DefineDataParser.java
+++ b/libs/natparse/src/main/java/org/amshove/natparse/parsing/DefineDataParser.java
@@ -173,7 +173,7 @@ public class DefineDataParser extends AbstractParser<IDefineData>
 					var currentRedefineNode = currentRedefine(variable);
 					if (currentRedefineNode != null)
 					{
-						while (mightBeFillerBytes(peek(1), peek(2)))
+						while (mightBeFillerAndItsBytes(peek(1), peek(2)))
 						{
 							parseRedefineFiller(currentRedefineNode);
 						}
@@ -480,9 +480,14 @@ public class DefineDataParser extends AbstractParser<IDefineData>
 		}
 	}
 
-	private boolean mightBeFillerBytes(SyntaxToken fillerToken, SyntaxToken maybeFillerBytes)
+	private boolean mightBeFillerAndItsBytes(SyntaxToken fillerToken, SyntaxToken maybeFillerBytes)
 	{
 		if (fillerToken == null || maybeFillerBytes == null)
+		{
+			return false;
+		}
+
+		if (fillerToken.kind() != SyntaxKind.FILLER)
 		{
 			return false;
 		}

--- a/libs/natparse/src/test/java/org/amshove/natparse/parsing/DefineDataParserShould.java
+++ b/libs/natparse/src/test/java/org/amshove/natparse/parsing/DefineDataParserShould.java
@@ -1012,6 +1012,24 @@ class DefineDataParserShould extends AbstractParserTest<IDefineData>
 	}
 
 	@Test
+	void notMistakeFillersOfFollowingRedefinesAsFillersOfItsOwn()
+	{
+		var defineData = assertParsesWithoutDiagnostics("""
+			DEFINE DATA
+			LOCAL
+			1 #ASTRING (A100)
+			1 REDEFINE #ASTRING
+				2 #SUBSTR (A10)
+				2 FILLER 50X
+			END-DEFINE
+			INPUT 50X 'A'
+			""");
+
+		var redefine = assertNodeType(defineData.variables().get(1), IRedefinitionNode.class);
+		assertThat(redefine.fillerBytes()).isEqualTo(50); // not 100 from FILLER 50X + INPUT 50X
+	}
+
+	@Test
 	void parseFillerInRedefinesInViews()
 	{
 		assertParsesWithoutDiagnostics("""

